src/main/groovy/io/cify/framework/core/Device.groovy

```diff
@@ -203,6 +213,12 @@ class Device implements IDevice {
                 stopRecording()
             }
 
```

Comment:
I think it would be better to do:
if (hasDriver()) {
    if (smth_else) {
    }
    ....
}

Reply:
Agree, fixed.